internal/logging/logger.go

```diff
@@ -30,7 +30,7 @@ func init() {
 
 func NewLogger(writer io.Writer) *logger {
 	return &logger{
-		zerolog.New(writer).With().Timestamp().Caller().Logger(),
+		Logger: zerolog.New(writer).Level(zerolog.InfoLevel).With().Timestamp().Caller().Logger(),
```

Comment:
That sounds like exactly what I want as the default.  Why do you say this is not what we want?

`SetLevel` sets the global level, which works the same way (drops all logs below that level).

We should really not support multiple loggers. Everything in our code base should ignore the global logger and only use `L`. So changing global shouldn't be different from changing the level of L, right?

Eventually I would still like to do #1609 and remove the static `L`.

Reply:
As in if you change the log level through `INFRA_LOG_LEVEL` to `debug` or `trace`, the lowest level will still be `info`. The logger this creates effectively has no `debug` or `trace` levels

Reply:
Ahh, because `SetLevel` does not change the level of this logger, right!  I think that's a problem with `SetLevel`. It should override this default, not the global setting that we don't really care about (because everything should use `L`).

I've pushed a commit which fixes this by changing `SetLevel`, to set the level of `L`.

Reply:
hmm, I'm not seeing `--log-level` working at all now, even when trying to set it to `warn`. I'll keep looking into why.

Reply:
I'm not sure it works like that since the logger will inherit the level of its parent, which would have Info here. You might end up need a brand new logger

Reply:
True, I think I did that with:

```go
L.Logger = L.Logger.Level(level)
```

But there are two other problems:
1. because we run a server in process calling `SetLevel` modifies the logger of both the server and the CLI, so the race detector noticed a data race. In production this wouldn't be a problem because there is no concurrency when `SetLevel` is called. #1609 would prevent this kind of thing because we wouldn't have a single static logger shared by everything in the process.
2. We call `UseServerLogger` after calling `SetLevel`, so we lose the level we previously set.

Setting the global level, or copying the log level in `UseServerLogger` / `UseFileLogger` are both options

Reply:
Changing to setting the global log level, because the data race is not going to be easy to fix.